### PR TITLE
FV1000 reader: fix preview names ordering

### PR DESCRIPTION
**Pull request by sbesson**
The usage of JDK 8 on various full repository jobs (Windows, breaking) highlighted some issues with the series 1 of some of the FV1000 filesets. Preliminary testing shows that the ordering of the `previewNames` was not fixed and depends on the parsing of the INI file. See https://trello.com/c/l2oMl30u/37-investigate-jdk-specific-automated-test-failures

This PR should address these ordering issues by:
- parsing the indexes of the preview/reference files from the INI keys into a `Map<Integer, String>` using the same strategy as the parsing of `filenames` and `roiFilenames`
- converting this map into a list of `previewNames`.

As multiple key prefixes where found across our sample files, a new static field `PREVIEWNAMES_PREFIX` is added to the review for future additions and the parsing includes a fallback to the default handling (unsorted addition to the `previewNames` list). Note this static field addition falls under the [serialization policy](https://github.com/openmicroscopy/design/issues/55) banner and can be reverted/modified depending on the decision made on this front.

A companion configuration PR will be opened to fix affected filesets in the data repository. For testing this PR:
- the full repository jobs should now pass against the `fv1000` format with various JDK implementations
- the affected FV1000 filesets should be reviewed - in particular the channel ordering in the second series should now match the first series

### CODE REVIEW

**Comment by sbesson** [comment]
As per https://ci.openmicroscopy.org/view/Failing/job/BIOFORMATS-DEV-breaking-full-repository/43/, moving out of breaking

File: components/formats-gpl/src/loci/formats/in/FV1000Reader.java [hunk 1]
```diff
@@ -480,6 +492,11 @@ else if (isPreviewName(value)) {
       }
     }
 
+    // Store sorted list of preview names
+    for (Integer key : previewFileNames.keySet()) {
```
Comment (mtbc) [inline components/formats-gpl/src/loci/formats/in/FV1000Reader.java]: is this loop just `previewNames.addAll(previewFileNames.values())`?

File: components/formats-gpl/src/loci/formats/in/FV1000Reader.java [hunk 2]
```diff
@@ -431,6 +436,8 @@ protected void initFile(String id) throws FormatException, IOException {
     pixelSize = new Double[NUM_DIMENSIONS];
 
     previewNames = new ArrayList<String>();
+    SortedMap<Integer, String> previewFileNames = new TreeMap<Integer, String>();
+    Integer previewIndex;
```
Comment (rleigh-codelibre) [inline components/formats-gpl/src/loci/formats/in/FV1000Reader.java]: Why is this declared outside the loop rather than at the single site of use?

**Comment by rleigh-codelibre** [comment]
Other than the two minor comments, the change looks reasonable.

**Comment by mtbc** [comment]
Good to merge.

**Comment by sbesson** [comment]
All automated tests successfully passed on [Linux/JDK7](https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-full-repository/277/), [Linux/JDK8](https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-breaking-full-repository/43/) and [Windows/JDK8](https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-full-repository-win/23/). Merging.

**Pull request merged**